File: deck/testdata/haskell/sample.hs
-- sample.hs — fixture for HaskellSegmenter's golden test.
module Main where

import Data.List (sort)

-- Color enumerates the three primary colors.
data Color = Red | Green | Blue

-- factorial computes n! for a non-negative n, by pattern-matching
-- equations that all stay in one card.
factorial :: Integer -> Integer
factorial 0 = 1
factorial n = n * factorial (n - 1)

main :: IO ()
main = do
  let xs = sort [3, 1, 2]
  print xs
  print (factorial 5)

File: deck/testdata/c/sample.c
#include <stdio.h>
#include <stdlib.h>

// Point is a 2D coordinate.
typedef struct {
	int x;
	int y;
} Point;

enum Color { RED, GREEN, BLUE };

int global_count;

int add(int a, int b) {
	return a + b;
}

int main(int argc, char **argv) {
	struct Point p = { 1, 2 };
	printf("{ %d, %d }\n", p.x, p.y);
	return 0;
}
